middleware/response.go

```diff
@@ -20,6 +20,7 @@ type badResponseLoggingWriter struct {
 	logBody       bool
 	bodyBytesLeft int
 	statusCode    int
+	writeError    error // The error returned when downstream Write() fails.
```

Comment:
Strictly this is the _last_ error.  I haven't investigated whether this matters.

Reply:
Yeah, you don't typically call `Write()` again after a previous call fails, so it should be fine?